massa-execution-worker/src/context.rs

```diff
@@ -567,24 +597,67 @@ impl ExecutionContext {
     /// * `to_addr`: optional crediting address (use None for pure coin destruction)
     /// * `amount`: amount of coins to transfer
     /// * `check_rights`: check that the sender has the right to spend the coins according to the call stack
+    /// * `current_slot`: necessary for check vesting (use None if from_addr = None)
     pub fn transfer_coins(
         &mut self,
         from_addr: Option<Address>,
         to_addr: Option<Address>,
         amount: Amount,
         check_rights: bool,
     ) -> Result<(), ExecutionError> {
-        // check access rights
-        if check_rights {
-            if let Some(from_addr) = &from_addr {
-                if !self.has_write_rights_on(from_addr) {
-                    return Err(ExecutionError::RuntimeError(format!(
-                        "spending from address {} is not allowed in this context",
-                        from_addr
+        if let Some(from_addr) = &from_addr {
+            // check access rights
+            if check_rights && !self.has_write_rights_on(from_addr) {
+                return Err(ExecutionError::RuntimeError(format!(
+                    "spending from address {} is not allowed in this context",
+                    from_addr
+                )));
+            }
+
+            // control vesting min_balance for sender address
+            if let Some(vesting_range) = self.find_vesting_range(from_addr, &self.slot) {
```

Comment:
todo: add a comment about the corner case of an address sending coins to itself

Reply:
if it send coins to itself we should skip the vesting check ?

Reply:
no, I think we can keep it, but a comment explaining that it's ok would be nice

Reply:
@modship can you make a followup issue on that ?

Reply:
https://github.com/massalabs/massa/issues/3699

Reply:
we could but it complicates things for no benefit. We could just explain in a comment that "sending coins to oneself implies spending the coins first, then receiving them. So the spending part can fail in the case of vesting"

Reply:
It's an edge case that should never happen. Adding the check would actually increase the processing in the nominal (non-self-sending) case